In [2]:
import numpy as np
#numpyの線形代数用ライブラリをLAとして定義
import numpy.linalg as LA
#例として以下の行列Aに対して固有値、固有ベクトルを求める
A = np.array([[3, 1], [0, 2]])
print(LA.eig(A))

(array([3., 2.]), array([[ 1.        , -0.70710678],
       [ 0.        ,  0.70710678]]))


<h1>Result</h1>
<p>When eigenvalue is $\lambda_1=3,$
        eigenvactor: $\boldsymbol{v_1} =
        \left[\begin{array}{c}
            1 \\
            0 \\
        \end{array}\right] \quad$</p>
<p>When eigenvalue is $\lambda_2=2,$
        eigenvactor: $\boldsymbol{v_1} =
        \left[\begin{array}{c}
            -0.70710678 \\
            0.70710678 \\
        \end{array}\right] \quad$</p>

<h1>結果</h1>
<p>固有値 $\lambda_1=3$のとき、固有ベクトル: 
        $\boldsymbol{v_1} =
        \left[\begin{array}{c}
            1 \\
            0 \\
        \end{array}\right] \quad$</p>
固有値 $\lambda_1=2$のとき、固有ベクトル:
        $\boldsymbol{v_2} =
        \left[\begin{array}{c}
            -0.70710678 \\
            0.70710678 \\
        \end{array}\right] \quad
$

In [3]:
#教科書に記載されている行列の例
A = np.array([[1, 2], [3, 4]])
LA.eig(A)

(array([-0.37228132,  5.37228132]), array([[-0.82456484, -0.41597356],
        [ 0.56576746, -0.90937671]]))

<h1>Result</h1>
As shown in the textbook, the eigenvalues of 
        $\boldsymbol{A} =
        \begin{equation*}
        \begin{bmatrix}
            1 & 2 \\
            3 & 4 \\
        \end{bmatrix}
        \end{equation*}$
        are $\lambda_1\approx5.37,$
        and $\lambda_2\approx-0.37$.

<h1>結果</h1>
本に記載の通り, 行列 
        $\boldsymbol{A} =
        \begin{equation*}
        \begin{bmatrix}
            1 & 2 \\
            3 & 4 \\
        \end{bmatrix}
        \end{equation*}$
        に対する固有値は $\lambda_1\approx5.37,$
        $\lambda_2\approx-0.37$である。